pywr-core/src/parameters/profiles/weekly.rs

```diff
@@ -0,0 +1,459 @@
+use crate::model::Model;
+use crate::parameters::{Parameter, ParameterMeta};
+use crate::scenario::ScenarioIndex;
+use crate::state::State;
+use crate::timestep::Timestep;
+use crate::PywrError;
+use std::any::Any;
+use time::{util::is_leap_year, Date};
+
+pub enum WeeklyInterpDay {
+    First,
+    Last,
+}
+
+// A weekly profile can be 52 or 53 week long
+enum WeeklyProfileSize {
+    FiftyTwo([f64; 52]),
+    FiftyThree([f64; 53]),
+}
+
+impl WeeklyProfileSize {
+    /// Get the week position in a calendar year from date. The position starts from 0 on the
+    /// first week day and ends with 1 on the last week day.
+    fn current_pos(&self, date: &Date) -> f64 {
```

Comment:
I wonder if we should use the ISO week number here? This would be a change from Pywr v1, but perhaps some consistency with a standard would be preferable than our own definition of a week. @Batch21 what do you think?

See: https://docs.rs/time/0.3.31/time/struct.Date.html#method.iso_week

Perhaps it could be a choice?